server/src/main/java/org/opensearch/snapshots/SnapshotShardsService.java

```diff
@@ -416,6 +417,10 @@ private void snapshot(
                             commitGeneration = lastRemoteUploadedIndexCommit.getGeneration();
                         } else {
                             wrappedSnapshot = indexShard.acquireLastIndexCommitAndRefresh(true);
+                            indexShard.waitForLocalCommitToBeUploadedToRemote(
+                                TimeValue.timeValueSeconds(5),
+                                TimeValue.timeValueMillis(100)
```

Comment:
this is very aggressive, we can have 5 minutes timeout and 1s polling interval with exponential backoff.